core/src/test/java/org/mobilitydata/gtfsvalidator/table/AnyTableLoaderTest.java

```diff
@@ -0,0 +1,216 @@
+package org.mobilitydata.gtfsvalidator.table;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.*;
+import static org.mobilitydata.gtfsvalidator.TestUtils.*;
+import static org.mockito.Mockito.*;
+
+import com.google.common.collect.ImmutableList;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import org.junit.Before;
+import org.junit.Test;
+import org.mobilitydata.gtfsvalidator.annotation.FieldLevelEnum;
+import org.mobilitydata.gtfsvalidator.input.CountryCode;
+import org.mobilitydata.gtfsvalidator.input.CurrentDateTime;
+import org.mobilitydata.gtfsvalidator.notice.*;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsStop;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsStopTableDescriptor;
+import org.mobilitydata.gtfsvalidator.validator.DefaultValidatorProvider;
+import org.mobilitydata.gtfsvalidator.validator.ValidationContext;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorLoader;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorProvider;
+
+public class AnyTableLoaderTest {
+
+  ValidationContext validationContext;
+
+  @Before
+  public void setup() {
+    validationContext =
+        ValidationContext.builder()
+            .setCountryCode(CountryCode.forStringOrUnknown("CA"))
+            .setCurrentDateTime(new CurrentDateTime(ZonedDateTime.now(ZoneId.systemDefault())))
+            .build();
+  }
+
+  @Test
+  public void invalidInputStream() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("_not_a_valid_file_");
+    NoticeContainer loaderNotices = new NoticeContainer();
+
+    AnyTableLoader.load(
+        testTableDescriptor,
+        mock(ValidatorProvider.class),
+        retrieveInputStream("_not_a_valid_file_"),
+        loaderNotices);
+
+    assertTrue(loaderNotices.hasValidationErrors());
+
+    assertValidationNotice(loaderNotices, CsvParsingFailedNotice.class);
+    assertEquals(1, loaderNotices.getValidationNotices().size());
+    verify(testTableDescriptor, times(0)).createContainerForHeaderAndEntities(any(), any(), any());
```

Comment:
I think line is maybe overkill.  I'd suggest instead you do something like:

```
GtfsTableContainer mockContainer = mock(GtfsTableContainer.class);
when(testTableDescriptor.createContainerForInvalidStatus(
              GtfsTableContainer.TableStatus.INVALID_HEADERS))
    .thenReturn(mockContainer);
```

and then verify the return value of `AnyTableLoader.load()` ala `assertThat(loadedContainer).isEqualTo(mockContainer);`

This has the benefit verify that the correct type of table was created for this situation.

Reply:
I agree with the change requested. I'm leaving one assertion as it's testing the invocation of the validators https://github.com/MobilityData/gtfs-validator/blob/0d62e9450d93348149e2fded4da9894742766265/core/src/test/java/org/mobilitydata/gtfsvalidator/table/AnyTableLoaderTest.java#L199. I'm open to suggestions if there is a better way, in this case, to guarantee that the execution of the validators.